### PR TITLE
gh-104773: PEP 594: Remove the ossaudiodev module

### PR DESCRIPTION
**Pull request by vstinner**
* Remove ossaudiodev extension in configure.ac and regenerate the configure script.
* Remove ossaudiodev in Modules/Setup and Modules/Setup.stdlib.in.

* Issue: gh-104773

----
:books: Documentation preview :books:: https://cpython-previews--104862.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
Using ``import ossaudiodev|from ossaudiodev`` regex, I only found a single project in PyPI top 5,000 projects (2023-04-13) using the ossaudiodev: **nltk**.

See [nltk/corpus/reader/timit.py](https://github.com/nltk/nltk/blob/13bf2bd4fc0ac76537ffb187e3eace4ea9cca26b/nltk/corpus/reader/timit.py#L436). This code tries ossaudiodev but catches properly ImportError and uses **pygame** if the ossaudiodev module is missing. If ossaudiodev and pygame are missing, it displays the error message:


```
you must install pygame or ossaudiodev for audio playback.
```

In the What's New In Python 3.13 entry, **I suggest replacing ossaudiodev with pygame**. IMO pygame is a more portable since it supports multiple audio playback backend depending on the platform, whereas ossaudiodev only supports OSS backend (ex: it doesn't support Windows).

File: Doc/whatsnew/3.13.rst [hunk 1]
```diff
@@ -157,6 +157,10 @@ Removed
   use the :mod:`subprocess` module instead.
   (Contributed by Victor Stinner in :gh:`104773`.)
 
+* :pep:`594`: Remove the :mod:`!ossaudiodev` module, deprecated in Python 3.11:
+  use the `pygame project <https://www.pygame.org/>`_ for audio playback.
```
Comment (hugovk) [inline Doc/whatsnew/3.13.rst:161]: Shall we also/instead mention the pygame-ce fork? https://pyga.me

More info: https://www.reddit.com/r/pygame/comments/1112q10/pygame_community_edition_announcement/

File: Lib/test/test_ossaudiodev.py
Comment (hugovk) [inline Lib/test/test_ossaudiodev.py:1]: Can we also delete `Lib/test/audiotest.au`?

**Review by hugovk — COMMENTED** [review]
Can these bits also be deleted?

```
configure
13492:    OSSAUDIODEV_LIBS="-lossaudio" ;; #(
13494:    OSSAUDIODEV_LIBS=""

configure.ac
3921:  [NetBSD*], [OSSAUDIODEV_LIBS="-lossaudio"],
3922:  [OSSAUDIODEV_LIBS=""]
```

**Comment by vstinner** [comment]
> Can these bits also be deleted? [OSSAUDIODEV_LIBS]

I created PR #104864 to remove it. Well spotted, I missed that one.

File: Lib/test/test_ossaudiodev.py
Comment (vstinner) [reply Lib/test/test_ossaudiodev.py:1]: My PR #104863 removes it.

File: Doc/whatsnew/3.13.rst (hunk 1 above)
Comment (vstinner) [reply Doc/whatsnew/3.13.rst:161]: I would prefer to only link to pygame for now. It's unclear to me where the community goes, but so far, "pygame" remains the reference project. I hope that the two communities will merge again. If not, we can update the doc later. The problem is that the Python documentation is not updated frequently.

**Pull request merged**